TileTowns/Assets/Scripts/Managers/RoadManager.cs

```
@@ -0,0 +1,189 @@
+        {
+            hasRoadRight = true;
+        }
+
+        if (_roadMap[ position + new Vector2Int(-1, 0)])
+        {
+            hasRoadLeft = true;
+        }
+        
+        if (hasRoadAbove && hasRoadBelow && hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position, upDownLeftRightRoad);
+        }
+        else if (hasRoadAbove && hasRoadBelow && hasRoadLeft)
+        {
+            SetRoad(position, upDownLeftRoad);
+        }
+        else if (hasRoadAbove && hasRoadBelow && hasRoadRight)
+        {
+            SetRoad(position, upDownRightRoad);
+        }
+        else if (hasRoadAbove && hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position, upLeftRightRoad);
+        }
+        else if (hasRoadBelow && hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position ,downLeftRightRoad);
+        }
+        else if (hasRoadAbove && hasRoadBelow)
+        {
+            SetRoad(position, upDownRoad);
+        }
+        else if (hasRoadAbove && hasRoadLeft)
+        {
+            SetRoad(position, upLeftRoad);
+        }
+        else if (hasRoadAbove && hasRoadRight)
+        {
+            SetRoad(position, upRightRoad);
+        }
+        else if (hasRoadBelow && hasRoadLeft)
+        {
+            SetRoad(position, downLeftRoad);
+        }
+        else if (hasRoadBelow && hasRoadRight)
+        {
+            SetRoad(position, downRightRoad);
+        }
+        else if (hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position, leftRightRoad);
+        }
+        else if (hasRoadAbove)
+        {
+            SetRoad(position, upRoad);
+        }
+        else if (hasRoadBelow)
+        {
+            SetRoad(position, downRoad);
+        }
+        else if (hasRoadLeft)
+        {
+            SetRoad(position, leftRoad);
+        }
+        else if (hasRoadRight)
+        {
+            SetRoad(position, rightRoad);
+        }
+        else
+        {
+            // No roads around
```

you can add InvalidOperationException or something (optional)

---

It's just a scenario, where it'a a lone building, don't think it should throw any exception

---

ow wait fuck xD
didnt read into it, you can write the `intentionally empty`, thats quite a good practice but however you prefer ;)